### PR TITLE
Merge `push to create`, `open PR from push`, and `push options` docs articles into one

### PR DESCRIPTION
**Pull request by lunny**
Add documentation for this feature.

gitea-push-hint

### CODE REVIEW

**Review by wolfogre — APPROVED** [review]

File: docs/content/doc/usage/push.en-us.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    identifier: "push"
+---
+
+**Table of Contents**
+
+{{< toc >}}
+
+There are some additional features when pushing commits to Gitea server.
+
+# Push Merge Hint
```
Comment (delvh) [inline docs/content/doc/usage/push.en-us.md]:
```suggestion
# Open PR through Push
```

File: docs/content/doc/usage/push.en-us.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+When you pushing commits to a non-default branch, you will get an information from
+Gitea which is a link, you can click the link and go to a compare page. It's a quick
+way to create a pull request or a code review yourself in the Gitea UI.
```
Comment (delvh) [inline docs/content/doc/usage/push.en-us.md]:
```suggestion
When you push commits to a non-default branch for the first time,
you will receive a link you can click on to visit the compare page of your branch compared to your main branch.
From there, it's easy to create a pull request, even if you want to target another branch.
```

**Review by delvh — APPROVED** [review]

**Comment by yardenshoham** [comment]
Can we somehow avoid referring to autogenerated files (`trans-copy`)?

Until today I was able to build the docs in Gitpod with
```bash
cd docs
make clean update
hugo server -D -F --baseUrl $(gp url 1313) --liveReloadPort=443 --appendPort=false --bind=0.0.0.0
```

I did it this way so no untracked (git) files would be created. Today, one of the pages is referring to an auto-generated page https://github.com/go-gitea/gitea/blob/ecf34fcd899fecad9782eea3097a4c38f9fe258b/docs/content/doc/usage/push.zh-tw.md?plain=1#L55.

It seems the ability to automatically build the docs without generating files that don't have to get cleaned up before a commit is made was lost

**Comment by lunny** [comment]
Don't know whether we should ref to doc without `.zh-tw` here.

**Pull request merged**